File: app/src/main/java/com/timotiusoktorio/pencake/ui/dialogs/DatePickerFragment.java
package com.timotiusoktorio.pencake.ui.dialogs;

import android.app.DatePickerDialog;
import android.app.Dialog;
import android.app.DialogFragment;
import android.content.Context;
import android.os.Bundle;

import java.util.Calendar;

public class DatePickerFragment extends DialogFragment {

    @Override
    public void onAttach(Context context) {
        super.onAttach(context);
        if (!(context instanceof DatePickerDialog.OnDateSetListener)) {
            throw new IllegalStateException(context.getClass().getSimpleName() +
                    " must implements DatePickerDialog.OnDateSetListener");
        }
    }

    @Override
    public Dialog onCreateDialog(Bundle savedInstanceState) {
        final Calendar calendar = Calendar.getInstance();
        int year = calendar.get(Calendar.YEAR);
        int month = calendar.get(Calendar.MONTH);
        int day = calendar.get(Calendar.DAY_OF_MONTH);
        return new DatePickerDialog(getActivity(), (DatePickerDialog.OnDateSetListener) getActivity(),
                year, month, day);
    }
}

File: app/src/main/java/com/timotiusoktorio/pencake/ui/dialogs/DatePickerFragment.kt
package com.timotiusoktorio.pencake.ui.dialogs

import android.app.DatePickerDialog
import android.app.Dialog
import android.app.DialogFragment
import android.content.Context
import android.os.Bundle
import java.util.*

class DatePickerFragment : DialogFragment() {

    override fun onAttach(context: Context) {
        super.onAttach(context)
        if (context !is DatePickerDialog.OnDateSetListener) {
            throw IllegalStateException("${context.javaClass.simpleName} must implements DatePickerDialog.OnDateSetListener")
        }
    }

    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
        val calendar = Calendar.getInstance()
        val year = calendar.get(Calendar.YEAR)
        val month = calendar.get(Calendar.MONTH)
        val day = calendar.get(Calendar.DAY_OF_MONTH)
        return DatePickerDialog(activity, activity as DatePickerDialog.OnDateSetListener, year, month, day)
    }
}